.github/actions/kots-e2e/action.yml

```diff
@@ -52,11 +55,66 @@ inputs:
     description: 'AWS region for uploading support bundle'
     default: 'us-east-1'
     required: false
+  gke-service-key:
+    description: 'Service key to authenticate a Google Cloud service account'
+    required: false
+  gke-cluster-region:
+    description: GKE cluster region to use
+    default: "us-west1"
+    required: false
+  gke-cluster-network:
+    description: GKE cluster network tag
+    default: "projects/replicated-qa/global/networks/default"
+    required: false
+  gke-cluster-subnetwork:
+    description: GKE cluster subnetwork tag
+    default: "projects/replicated-qa/regions/us-west1/subnetworks/default"
+    required: false
+  gke-project:
+    description: Google Cloud project name
+    default: "replicated-qa"
+    required: false
+
 runs:
   using: "composite"
   steps:
+    - name: Generate GKE cluster name
+      if: inputs.platform == 'gke'
+      id: cluster-name
+      shell: bash
+      run: |
+        echo "cluster_name=kots-ci-${{ github.run_id }}-${{ github.run_attempt }}-$(echo '${{ github.job }}' | cut -c -18)" >> $GITHUB_OUTPUT
+      
+    - name: Authenticate to Google Cloud
+      if: inputs.platform == 'gke'
+      uses: google-github-actions/auth@v1
+      with:
+        credentials_json: ${{ inputs.gke-service-key }}
+
+    - name: Setup Cloud SDK
+      if: inputs.platform == 'gke'
+      uses: google-github-actions/setup-gcloud@v1
+      with:
+        install_components: 'gke-gcloud-auth-plugin'
+
+    - name: Create the GKE cluster
+      if: inputs.platform == 'gke'
+      shell: bash
+      run: |
+        gcloud container clusters create-auto "${{ steps.cluster-name.outputs.cluster_name }}" --project "${{ inputs.gke-project }}" --region "${{ inputs.gke-cluster-region }}" --network "${{ inputs.gke-cluster-network }}"  --subnetwork "${{ inputs.gke-cluster-subnetwork }}"
+
+    - name: Get GKE cluster credentials for kubectl
+      if: inputs.platform == 'gke'
+      uses: google-github-actions/get-gke-credentials@v1
+      with:
+        cluster_name: ${{ steps.cluster-name.outputs.cluster_name }}
+        location: ${{ inputs.gke-cluster-region }}
+        project_id: ${{ inputs.gke-project }}
+
```

Comment:
Could this be its own action and it's called from here?  Can be maintained in this repo or outside, but would simplify the logic in case we add more platforms down the road.

Something like:
```
- name: Create a GKE cluster
  if: inputs.platform == 'gke'
  uses: ./.github/actions/clusters/gke/create
  with:
    ...
- name: Destroy a GKE cluster
  if: ${{ always() && inputs.platform == 'gke' }}
  uses: ./.github/actions/clusters/gke/destroy
  with:
    ...
```

Reply:
+1

Reply:
😂 I originally had the gke stuff in it's own action but decided that was making things overly complicated. But I do like the separation, so the gke action file is back